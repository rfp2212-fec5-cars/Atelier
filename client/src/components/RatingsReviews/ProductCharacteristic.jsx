import React from 'react';

var ProductCharacteristic = ({ type, info }) => {
  let left, middle, right;
  if (type === 'Comfort' || type === 'Quality') {
    left = 'poor';
    middle = '';
    right = 'perfect';
  } else {
    left = 'too small';
    middle = 'perfect';
    right = 'too large';
  }
  //the length of the whole 3 bar is 250px
  //we minus 8 because if info.value is 5, 定位符的最左侧指向bar的尾端，而不是我们想看到的定位符中间,8是实验来的
  let moving = (info.value / 5) * 250 - 8;
  return (
    <div className='characteristic'>
      <div>{type}</div>
      <div>
        <div className='pointer' style={{ marginLeft: `${moving + 'px'}` }}>▼</div>
        <div className='left'></div>
        <div className='middle'></div>
        <div className='right'></div>
      </div>

      <div className='producttext'>
        <div className='lefttext'>{left}</div>
        <div className='middletext'>{middle}</div>
        <div className='righttext'>{right}</div>
      </div>
    </div>

  );
};
export default ProductCharacteristic;