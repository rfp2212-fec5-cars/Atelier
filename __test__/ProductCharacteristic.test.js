import TestRenderer from 'react-test-renderer';
import ProductCharacteristic from '../client/src/components/RatingsReviews/ProductCharacteristic.jsx';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { render, screen, fireEvent } from '@testing-library/react';



describe('sortStar testing', () => {
  it('should exists', () => {
    const testRenderer = TestRenderer.create(
      <ProductCharacteristic type={'Comfort'} info = {{'value': 4}}></ProductCharacteristic>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have 11 div', () => {
    const testRenderer = TestRenderer.create(
      <ProductCharacteristic type={'Comfort'} info = {{'value': 4}}></ProductCharacteristic>
    );
    const testInstance = testRenderer.root;
    expect(testInstance.findAll(node => node.type === 'div')).toHaveLength(11);
  });
  it('should have Comfort text', () => {
    const {getByText} = render(
      <ProductCharacteristic type={'Comfort'} info = {{'value': 4}}></ProductCharacteristic>
    );
    expect(getByText('Comfort')).toBeTruthy();
    expect(getByText('poor')).toBeTruthy();
    expect(getByText('perfect')).toBeTruthy();

  });

});


