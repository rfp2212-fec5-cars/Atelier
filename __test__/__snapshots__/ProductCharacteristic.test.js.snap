// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortStar testing should exists 1`] = `
<div
  className="characteristic"
>
  <div>
    Comfort
  </div>
  <div>
    <div
      className="pointer"
      style={
        {
          "marginLeft": "192px",
        }
      }
    >
      â–¼
    </div>
    <div
      className="leftbar"
    />
    <div
      className="middlebar"
    />
    <div
      className="rightbar"
    />
  </div>
  <div
    className="producttext"
  >
    <div
      className="bar-text"
    >
      poor
    </div>
    <div
      className="bar-text"
    />
    <div
      className="bar-text"
    >
      perfect
    </div>
  </div>
</div>
`;
